### PR TITLE
remove LogRecord.name references

### PR DESCRIPTION
**Pull request by codeboten**
The name field is deprecated. Removing references to it.

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by codeboten** [comment]
Fixing the tests.

File: exporter/splunkhecexporter/logdata_to_splunk.go [hunk 1]
```diff
@@ -41,12 +41,8 @@ func mapLogRecordToSplunkEvent(res pdata.Resource, lr pdata.LogRecord, config *C
 	sourceTypeKey := config.HecToOtelAttrs.SourceType
 	indexKey := config.HecToOtelAttrs.Index
 	hostKey := config.HecToOtelAttrs.Host
-	nameKey := config.HecFields.Name
 	severityTextKey := config.HecFields.SeverityText
 	severityNumberKey := config.HecFields.SeverityNumber
-	if lr.Name() != "" {
-		fields[nameKey] = lr.Name()
-	}
```
Comment (tigrannajaryan) [inline exporter/splunkhecexporter/logdata_to_splunk.go:49]: 👍 

I forgot to remove this in my cleanup.

**Review by tigrannajaryan — APPROVED** [review]
FYI @atoulme

**Review by tigrannajaryan — COMMENTED** [review]
@dmitryax please also have a look.

**Review by dmitryax — APPROVED** [review]

File: exporter/splunkhecexporter/client_test.go [hunk 2]
```diff
@@ -147,8 +147,8 @@ func createLogDataWithCustomLibraries(numResources int, libraries []string, numR
 			for k := 0; k < numRecords[j]; k++ {
 				ts := pdata.Timestamp(int64(k) * time.Millisecond.Nanoseconds())
 				logRecord := ill.LogRecords().AppendEmpty()
-				logRecord.SetName(fmt.Sprintf("%d_%d_%d", i, j, k))
 				logRecord.Body().SetStringVal("mylog")
+				logRecord.Attributes().InsertString(splunk.DefaultNameLabel, fmt.Sprintf("%d_%d_%d", i, j, k))
```
[the remark this answers is not in the document]
Comment (codeboten) [reply exporter/splunkhecexporter/client_test.go:151]: I'll address that in a separate PR, I was seeing various tests fail without the distinct attribute.

**Pull request merged**